core/src/main/csharp/ch/cyberduck/core/SystemCertificateStore.cs

```diff
@@ -186,24 +187,21 @@ public bool verify(CertificateTrustCallback prompt, String hostName, List certs)
             return true;
         }
 
-        public static IList<string> ListAliases()
+        public static IReadOnlyCollection<string> ListAliases()
         {
-            X509Store store = new X509Store(StoreName.My, StoreLocation.CurrentUser);
-            IList<string> certs = new List<string>();
-            try
+            using X509Store store = new(StoreName.My, StoreLocation.CurrentUser);
+            HashSet<string> certs = new();
+            store.Open(OpenFlags.ReadOnly);
+            foreach(X509Certificate2 certificate in store.Certificates)
             {
-                store.Open(OpenFlags.ReadOnly);
-                foreach (X509Certificate2 certificate in store.Certificates)
+                var alias = string.IsNullOrWhiteSpace(certificate.FriendlyName)
+                    ? certificate.GetNameInfo(X509NameType.SimpleName, false)
+                    : certificate.FriendlyName;
+                if(!certs.Add(alias) && Log.isDebugEnabled())
                 {
-                    certs.Add(String.IsNullOrEmpty(certificate.FriendlyName)
-                        ? certificate.GetNameInfo(X509NameType.SimpleName, false)
-                        : certificate.FriendlyName);
+                    Log.debug($"Skipping duplicate alias \"{alias}\"");
                 }
             }
-            finally
-            {
-                store.Close();
-            }
```

Comment:
Closing the store is no longer required?

Reply:
Using / Dispose-Pattern does this automatically.